compiler/tests-spec/testData/diagnostics/notLinked/dfa/pos/53.fir.kt

```diff
@@ -33,7 +33,7 @@ fun case_3(x: Int) = ""
 fun case_3(x: Int?) = 10
 fun case_3() {
     if (case_3_prop != null) {
-        val z = case_3(<!DEBUG_INFO_EXPRESSION_TYPE("kotlin.Int?")!>case_3_prop<!>)
+        val z = case_3(<!DEBUG_INFO_EXPRESSION_TYPE("kotlin.Int? & kotlin.Int?")!>case_3_prop<!>)
```

Comment:
Can you please change `DEBUG_INFO_EXPRESSION_TYPE` reporter so it won't render types from unstable smartcasts?

Reply:
I think this is the place: https://github.com/JetBrains/kotlin/blob/master/compiler/tests-common-new/tests/org/jetbrains/kotlin/test/frontend/fir/handlers/FirDiagnosticsHandler.kt#L179 ?

Reply:
Thanks! I have updated #4382.